cloud/openstack/os_auth.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/python
+
+# Copyright (c) 2015 Hewlett-Packard Development Company, L.P.
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+try:
+    import shade
+    from shade import meta
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+DOCUMENTATION = '''
+---
+module: os_auth
+short_description: Retrieve an auth token
+extends_documentation_fragment: openstack
+description:
+   - Retrieve an auth token from an OpenStack Cloud
+options:
+  state:
+    description:
+      - Should the resource be present or absent.
+    choices: [present, absent]
+    default: present
+requirements: ["shade"]
+'''
+
+EXAMPLES = '''
+# Fetch facts about an instance called vm1
+- os_auth:
+    cloud: rax-dfw
+  register: results
+- debug: var=results
+'''
+
+def main():
+
+    argument_spec = openstack_full_argument_spec(
+        state=dict(default='present', choices=['absent', 'present']),
+    )
+    module_kwargs = openstack_module_kwargs()
+    module = AnsibleModule(argument_spec, **module_kwargs)
+
+    if not HAS_SHADE:
+        module.fail_json(msg='shade is required for this module')
+
+    try:
+        cloud = shade.openstack_cloud(**module.params)
+        module.exit_json(
+            changed=False,
+            auth_token=cloud.auth_token,
+            service_catalog=cloud.service_catalog)
```

Comment:
Instead of just dumping the token and catalog in the return to be registered, why not make this an actual fact module http://docs.ansible.com/developing_modules.html#module-provided-facts and automatically make facts out of the data? Saves a step in playbooks.